topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,423 @@
-[//]: # (title: Integrate KMM into an existing application)
+[//]: # (title: Make your Android application work on iOS – tutorial)
 [//]: # (auxiliary-id: Integrate_KMM_into_an_existing_application)
 
-Here you can learn how to integrate Kotlin Multiplatform Mobile (KMM) into an existing Android application and make it multiplatform so that it works on both Android and iOS. 
-You will understand what kind of code is good to share and how to use shared code in your existing application.
+Here you can learn how to make your existing Android application cross platform so that it works both on Android and iOS. 
+You'll be able to write code and test it for Android and iOS only once, in one place.
 
-Sharing code saves you time and effort on writing code and testing it for Android and iOS as you only need to do it once, in the one place.
+This tutorial uses a [sample Android application](https://github.com/Kotlin/kmm-integration-sample) with a single screen for entering a username and password. 
+The credentials are validated and saved to an in-memory database.
 
-Before you begin, learn how to [create and configure a KMM application](create-first-app.md).
+If you aren't familiar with KMM, learn how to [create and configure a KMM application from scratch](create-first-app.md).
 
-## Decide what to share
+## Prepare an environment for development
 
-An application includes two layers:
+1. [Set up your environment for KMM development](setup.md) by installing the necessary tools on a macOS.
 
-* The **frontend** that defines the user interface and its behavior, including animations and transitions.
-* The **backend** that defines the business logic, data management, network stack, and other components that support the frontend.
+   >* For completing specific steps in this tutorial you need a Mac with macOS. These steps include writing iOS-specific code and running an iOS application.  
```

Comment:
It seems that the info here is too important to be inside the note:
- The info that you should have Mac with macOS should be earlier (or not wrapped in the note).
- If Android Studio 4.2 is a requirement, it shouldn't be in a note. 

I'd prefer to see in that way:

## Prepare an environment ...

To start the project, you should have a Mac with macOS to work with Xcode...

1. Set up your env. This tutorial requires AS 4.2.
 or 

2. Update your Android studio up to 4.2.